public/app/types/datasources.ts

```diff
@@ -21,6 +22,7 @@ export interface DataSourceSettingsState {
   testingStatus?: {
     message?: string | null;
     status?: string | null;
+    details?: HealthCheckResult['details'];
```

Comment:
Maybe `testingStatus` deserves its own interface/type so we get rid of all `DataSourceSettingsState['testingStatus']` types.